src/compiler/types.ts

```diff
@@ -6125,6 +6125,8 @@ export const enum TypeFlags {
     NonPrimitive    = 1 << 26,  // intrinsic object type
     TemplateLiteral = 1 << 27,  // Template literal type
     StringMapping   = 1 << 28,  // Uppercase/Lowercase type
+    /** @internal */
+    Reserved1       = 1 << 29,  // Used by union/intersection type construction

```

Comment:
This is kind of funky - I'd just directly make this `IncludesConstrainedTypeVariable`.